docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/AutoCounter.rst

```diff
@@ -16,28 +16,38 @@ simulation (unlike target-level performance counters), these counters do not
 affect the behavior of the simulated machine, no matter how often they are
 sampled.
 
-Ad-hoc Performance Counters
-------------------------------
+Chisel Interface
+----------------
 
 AutoCounter enables the addition of ad-hoc counters using the ``PerfCounter``
-function.  The ``PerfCounter`` function takes 3 arguments: A UInt signal encoding an event increment, a counter label, and the counter description. The increment indicates the number of times the event of occurs in a single cycle and will be added on every simulated cycle to the generated performance counter. An increment can be boolean, with true implying the event occured (once) in the cycle, or multibit, to capture events that occur more than once in a cycle. We show two examples
-of using the ``PerfCounter`` function below.
+object in the `midas.targetutils` package. PerfCounters counters can be added in one of two modes:
+
+# `Incrementing`, using ``PerfCounter.apply``. Here a boolean or multi-bit value is accumulated into a 64b register.
+# `Identity`, using``PerfCounter.identity``. Here the user provided is sampled directly. This can be used
+  to annotate a sample with values are not accumulator-like (e.g., a PC),
+  and permits the user to define more complex instrumentation logic in the target itself.
+
+We give examples of using PerfCounter below:
 
 .. code-block:: scala
 
     midas.targetutils.PerfCounter(en_clock, "gate_clock", "Core clock gated")
     // A multibit example
```

Comment:
Nit: In the past I got confused by what it means to add a multi-bit signal (I had to dig into the xform for the solution). For example, if `insns_ret == b0100` did it mean that `b0100` interpreted as an `UInt` would be added to the accumulation register or would a "popcount"ed version be added (i.e. `b0100` converts to "add 1 to accumulation register"). Maybe some clarity on this would be nice (maybe just stating this example in the comment of the multibit example).